### PR TITLE
fix(graphql-relational-transformer): properly create sortkey for generated gsi on related table

### PR DESCRIPTION
**Pull request by atierian**
#### Description of changes
Adds missing support for sort keys on generated GSIs on the related model DDB table in the new `references` world.

Example schema
```graphql
type Foo @model {
  id: ID! @primaryKey(sortKeyFields: ["baz", "quux"])
  baz: String!
  quux: Int!
  bar: Bar @hasOne(references: ["fooID", "fooBaz", "fooQuux"])
}

type Bar @model {
  id: ID!
  fooID: ID
  fooBaz: String
  fooQuux: Int
  foo: Foo @belongsTo(references: ["fooID", "fooBaz", "fooQuux"])
}
```

##### CDK / CloudFormation Parameters Changed
None

#### Issue #, if available
N/A

#### Description of how you validated changes
Unit and manual E2E testing

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] ~Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)~
- [ ] ~Relevant documentation is changed or added (and PR referenced)~
- [ ] ~New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies~
- [ ] ~Any CDK or CloudFormation parameter changes are called out explicitly~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by AaronZyLee — APPROVED** [review]
LGTM. Can we add unit tests for this function? (Covering the three scenarios, can be addressed separately)

File: packages/amplify-graphql-relational-transformer/src/resolvers.ts [hunk 1]
```diff
@@ -61,8 +61,9 @@ export const updateTableForReferencesConnection = (
   // -- ideally further up the chain
   const partitionKeyType = attributeTypeFromType(referenceNode.type, ctx);
 
-  // TODO: Add support for sortKey in GSI
-  const sortKey = undefined;
+  // The sortKeys are any referencesNodes following the first element.
+  const sortKeyReferenceNodes = referenceNodes.slice(1);
```
Comment (phani-srikar) [inline packages/amplify-graphql-relational-transformer/src/resolvers.ts:65]: what if this doesn't exist? Would add some unit tests to cover case of none, one and more than one sort key fields.

Comment (phani-srikar) [reply packages/amplify-graphql-relational-transformer/src/resolvers.ts:65]: Referring to the later, but taking a deeper look, the `addGlobalSecondaryIndex` does handle [undefined sortKey](https://github.com/aws-amplify/amplify-category-api/blob/162e83dd9ff34c6c72b4cffe3ca7059fb235b303/packages/amplify-graphql-relational-transformer/src/resolvers.ts#L175).

**Review by phani-srikar — APPROVED** [review]
